### DISCUSSION TITLE
按照官方文档步骤操作，启动session超时

### DISCUSSION THREAD

**Discussion by nanyoullm** [alibaba/GraphScope#1551] [Q&A]
环境：win10 wsl2 python3.9.7
按照官方文档部署kind后，如图
<img width="781" alt="企业微信截图_16524144558574" src="https://user-images.githubusercontent.com/13297712/168208707-36c7fc8a-a22f-4504-8851-cb896ad99ccf.png">
拷贝官方代码在jupyter notebook启动session后超时报错（如下），请问是哪一步出错了呢？手动新建了~/test_data/目录
------------------------------------------------------------
TimeoutError                              Traceback (most recent call last)
~/anaconda3/lib/python3.9/site-packages/graphscope/deploy/kubernetes/cluster.py in start(self)
    568             time.sleep(1)
--> 569             self._waiting_for_services_ready()
    570             self._coordinator_endpoint = self._get_coordinator_endpoint()

~/anaconda3/lib/python3.9/site-packages/graphscope/deploy/kubernetes/cluster.py in _waiting_for_services_ready(self)
    495 
--> 496         if wait_for_deployment_complete(
    497             api_client=self._api_client,

~/anaconda3/lib/python3.9/site-packages/graphscope/deploy/kubernetes/utils.py in wait_for_deployment_complete(api_client, namespace, name, timeout_seconds)
    136                         raise K8sError("Deployment {} start failed.".format(name))
--> 137     raise TimeoutError("Waiting timeout for deployment {}".format(name))
    138 

TimeoutError: Waiting timeout for deployment coordinator-jnnytb

The above exception was the direct cause of the following exception:

K8sError                                  Traceback (most recent call last)
/tmp/ipykernel_20572/872928060.py in <module>
     16 }
     17 
---> 18 sess = graphscope.session(k8s_volumes=k8s_volumes)

~/anaconda3/lib/python3.9/site-packages/graphscope/client/utils.py in wrapper(*args, **kwargs)
    355             func.__defaults__ = tuple(new_defaults)
    356 
--> 357             return_value = func(*args, **kwargs)
    358 
    359             # Restore original defaults.

~/anaconda3/lib/python3.9/site-packages/graphscope/client/session.py in __init__(self, config, addr, mode, cluster_type, num_workers, preemptive, k8s_namespace, k8s_service_type, k8s_gs_image, k8s_etcd_image, k8s_dataset_image, k8s_image_pull_policy, k8s_image_pull_secrets, k8s_coordinator_cpu, k8s_coordinator_mem, etcd_addrs, k8s_etcd_num_pods, k8s_etcd_cpu, k8s_etcd_mem, k8s_vineyard_daemonset, k8s_vineyard_cpu, k8s_vineyard_mem, vineyard_shared_mem, k8s_engine_cpu, k8s_engine_mem, k8s_mars_worker_cpu, k8s_mars_worker_mem, k8s_mars_scheduler_cpu, k8s_mars_scheduler_mem, k8s_volumes, k8s_waiting_for_delete, timeout_seconds, dangling_timeout_seconds, with_mars, mount_dataset, reconnect, **kw)
    713         # create and connect session
    714         with CaptureKeyboardInterrupt(self.close):
--> 715             self._connect()
    716 
    717         self._disconnected = False

~/anaconda3/lib/python3.9/site-packages/graphscope/client/session.py in _connect(self)
   1038         # launching graphscope service
   1039         if self._launcher is not None:
-> 1040             self._launcher.start()
   1041             self._coordinator_endpoint = self._launcher.coordinator_endpoint
   1042 

~/anaconda3/lib/python3.9/site-packages/graphscope/deploy/kubernetes/cluster.py in start(self)
    577             self._dump_coordinator_failed_status()
    578             self.stop()
--> 579             raise K8sError(
    580                 "Error when launching Coordinator on kubernetes cluster"
    581             ) from e

K8sError: Error when launching Coordinator on kubernetes cluster

**Comment by nanyoullm** [comment]
sess = graphscope.session(num_workers=2)
这样配置是可以成功运行

**Reply by nanyoullm** [reply]
k8s_volumns中去掉
        "mounts": {
            "mountPath": "/testingdata"
        }
后，是可以启动session，看起来是这一项配置导致超时

**Reply by nanyoullm** [reply]
@yecol 大佬有空能否帮忙看看

**Reply by acezen** [reply]
这个mountPath需要指定的是你的test目录在host上的实际位置，不一定是/tesingdata, 根据你的情况，应该是~/test_data, “~”需要替换为HOME目录实际的值

**Reply by acezen** [reply]
> sess = graphscope.session(num_workers=2) 这样配置是可以成功运行
请问更改mountPath后还有问题吗？

**Reply by nanyoullm** [reply]
> 
因为我也不熟悉mount的配置，还在学习。试了几种还是不行呢，你刚刚说的配置是如下吗？

k8s_volumes = {
    "data": {
        "type": "hostPath",
        "field": {
            "path": os.path.expanduser("\~/test_data"),
            "type": "Directory"
        },
        "mounts": {
            "mountPath": "~/test_data"
        }
    }
}

pod里的路径和linux的路径是保持一致的吗

**Reply by acezen** [reply]
是的，不过"mountPath": "~/test_data" 里的 ~ 可以替换成你的实际目录，比如/home/xxx

**Reply by nanyoullm** [reply]
@acezen 谢谢，试了下，在客户端还是启动session超时，现在我在尝试多机k8s集群上的部署了
